plutus-tx/src/PlutusTx/AssocMap.hs

```diff
@@ -53,15 +53,13 @@ import GHC.Generics (Generic)
 import Language.Haskell.TH.Syntax as TH (Lift)
 import Prettyprinter (Pretty (..))
 
-{- HLINT ignore "Use newtype instead of data" -}
-
 -- See Note [Optimising Value].
 -- | A 'Map' of key-value pairs.
 newtype Map k v = Map {unMap :: [(k, v)]}
```

Comment:
Can you expand the Haddock of the `Map` type to summarize which operations are unsafe? Otherwise people may miss the Haddock on the instances.